spec/models/user_spec.rb

```diff
@@ -0,0 +1,61 @@
+describe User do
+
+  context "valid user" do
+    let(:user) { User.create(username: "Jeff", email: "rogerthat@example.com", password: "intriguing") }
+    it "has a username" do
+
+      expect(user.username).to eq("Jeff")
+    end
+
+    it "has an email" do
+
+      expect(user.email).to eq("rogerthat@example.com")
+    end
+
+    it "has a password" do
+
+      expect(user.password).to eq("intriguing")
+    end
+
+    it "validates  the user" do
+      expect(User.authenticate(email: user.email, password: "intriguing")).to eq(user)
```

Comment:
This test setup is kind of confusing. I'd prefer to see a separate context for your authenticate method, and then one test that `it "returns the user for authentication with valid credentials"` and one that `it "returns nil for authentication with invalid credentials"`.